src/shape.cpp

```diff
@@ -676,15 +677,12 @@ void migraphx_from_value(const value& v, shape& s)
             auto v_dd = v.at("dynamic_dimensions");
             std::vector<shape::dynamic_dimension> dyn_dims(v.at("dynamic_dimensions").size());
             std::transform(v_dd.begin(), v_dd.end(), dyn_dims.begin(), [](migraphx::value x) {
-                auto x_min  = x.at("min").template to<size_t>();
-                auto x_max  = x.at("max").template to<size_t>();
-                auto v_opts = x.at("opts");
-                std::set<size_t> set_x_opts;
-                std::transform(v_opts.begin(),
-                               v_opts.end(),
-                               std::inserter(set_x_opts, set_x_opts.begin()),
-                               [](auto v_opt) { return v_opt.template to<size_t>(); });
-                return shape::dynamic_dimension{x_min, x_max, set_x_opts};
+                auto x_min      = x.at("min").template to<size_t>();
+                auto x_max      = x.at("max").template to<size_t>();
+                auto v_optimals = x.at("optimals");
+                std::set<size_t> set_x_optimals =
+                    from_value<std::set<std::size_t>>(x.at("optimals"));
+                return shape::dynamic_dimension{x_min, x_max, set_x_optimals};
```

Comment:
Actually `dynamic_dimension` has a reflect method so this could just be `return from_value<shape::dynamic_dimension>(x)`.